src/nnet3/nnet-utils.cc

```diff
@@ -40,6 +40,106 @@ int32 NumInputNodes(const Nnet &nnet) {
   return ans;
 }
 
+int32 GetTModulusForIvector(const Nnet &nnet) {
+  std::vector<std::string> name;
+  name.push_back("ivector");
+  std::vector<int32> value;
+  value.resize(name.size(), 0);
+  for (int32 i = 0; i < nnet.NumNodes(); i++) {
+    const NetworkNode &node = nnet.GetNode(i);
+    // skip output node to circumvent the problem when a recurrent output 
+    // is "label delayed" and thus also contain keyword "Offset", which is not
+    // what we are trying to look at for our "offset"  
+    if (node.node_type == kDescriptor) {
+      for (int32 p = 0; p < node.descriptor.NumParts(); p++) {
+        const SumDescriptor &this_part = node.descriptor.Part(p);
+        IntoSumDescriptor(nnet, this_part, name, &value);
+      }
+    }
+  }
+  return value[0]; 
+}
+
+void IntoSumDescriptor(const Nnet &nnet,
```

Comment:
Yes- also, in the google style guide, function names must be verbs not
nouns.
so vijay's suggestion is more in line with that.

On Thu, Mar 10, 2016 at 9:20 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In src/nnet3/nnet-utils.cc
> <https://github.com/kaldi-asr/kaldi/pull/468#discussion_r55781874>:
>
> > +  for (int32 i = 0; i < nnet.NumNodes(); i++) {
> > +    const NetworkNode &node = nnet.GetNode(i);
> > +    // skip output node to circumvent the problem when a recurrent output
> > +    // is "label delayed" and thus also contain keyword "Offset", which is not
> > +    // what we are trying to look at for our "offset"
> > +    if (node.node_type == kDescriptor) {
> > +      for (int32 p = 0; p < node.descriptor.NumParts(); p++) {
> > +        const SumDescriptor &this_part = node.descriptor.Part(p);
> > +        IntoSumDescriptor(nnet, this_part, name, &value);
> > +      }
> > +    }
> > +  }
> > +  return value[0];
> > +}
> > +
> > +void IntoSumDescriptor(const Nnet &nnet,
>
> The name IntoSumDescriptor is not descriptive. Better use some name like
> ExtractOffsetsFromSumDescriptor(). Rename the variable values too.
>
> Same thing goes for the other descriptors too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/468/files#r55781874>.
>